src/metabase/driver/postgres.clj

```diff
@@ -263,6 +267,10 @@
 ;;; |                                           metabase.driver.sql impls                                            |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defmethod sql.qp/current-datetime-honeysql-form :postgres
+  [_driver]
+  (hx/with-database-type-info :%now "timestamptz"))
```

Comment:
Not part of this fix but Postgres `now()` returns a `timestamp with time zone` so I added this impl to include the type info so we generate `now()` in SQL instead of `cast(now() AS timestamp)` which is not necessary in Postgres land.